### ISSUE TITLE
Is 'batch.Write' atomic?

### ISSUE THREAD

**Issue by baxiry** [tidwall/wal#30]
Is 'batch.Write' atomic?
Thanks!

**Comment by tidwall** [comment]
The `batch.Write` call is just a temporary in-memory store that queues each entry until log.WriteBatch is called.
But `log.WriteBatch` is atomic.

**Comment by baxiry** [comment]
Thanks

**Comment by RangerCD** [comment]
No, `Batch.Write` is not atomic, and parallel `Batch.Write` might cause data corruption. You should create separate batch for each goroutine, or just call `Log.Write` directly if only single entry in each goroutine.

**Comment by baxiry** [comment]
@RangerCD  thanks
